polynote-server/src/main/scala/polynote/server/NotebookSession.scala

```diff
@@ -37,9 +37,11 @@ class NotebookSession(subscriber: KernelSubscriber, streamingHandles: StreamingH
       subscriber.checkPermission(Permission.ModifyNotebook) *> subscriber.update(update)
 
     case RunCell(ids) =>
-      if (ids.isEmpty) ZIO.unit else {
-        ids.map(id => subscriber.checkPermission(Permission.ExecuteCell(_, id))).reduce(_ *> _) *>
-          ids.map(id => subscriber.publisher.queueCell(id)).sequence.flatMap(_.sequence).unit
+      ZIO.unless(ids.isEmpty) {
```

Comment:
This was the problem... the inner task (e.g the `fiber.join.onInterrupt(whatever)`) never actually got evaluated. The task runs anyway, independently of that, but in order to see any finalizers that happen on `fiber.join.doSomeFinalizers(...)` that effect would have to be evaluated.

Reply:
(it's much easier to see that it gets evaluated when the `.sequence` calls are replaced with `foreach` calls 🤷